src/chainstate/stacks/index/storage.rs

```diff
@@ -1171,11 +871,7 @@ impl TrieFileStorage {
                 //   I *think* this is no longer necessary in the fork-table-less construction.
                 //   the back_pointer's consensus bytes uses this block_hash instead of a back_block
                 //   integer. This means that it would _always_ be included the node's hash computation.
-                let block_hash = block_map.get_block_header_hash(ptr.back_block())
-                    .ok_or_else(|| {
-                        error!("Failed to look up block at {}", ptr.back_block());
-                        Error::NotFoundError
-                    })?;
+                let block_hash = map.get_block_hash_caching(ptr.back_block())?;
```

Comment:
Is the above comment still relevant?

Reply:
Yes, I think so. What happens here is that the same "back block hash" is getting read into a node's hasher twice -- once when it includes it's pointer's consensus bytes, and once more here (because a back pointer's consensus bytes don't include the back_block pointer, but instead include that block's hash).

Reply:
Gotcha, thanks!